crates/starknet-devnet/tests/test_abort_blocks.rs

```diff
@@ -0,0 +1,287 @@
+        let first_block = &devnet
+            .send_custom_rpc("starknet_getBlockWithTxHashes", json!({ "block_id": "latest" }))
+            .await["result"];
+
+        devnet.post_json("/create_block".into(), Body::from(json!({}).to_string())).await.unwrap();
+        let second_block = &devnet
+            .send_custom_rpc("starknet_getBlockWithTxHashes", json!({ "block_id": "latest" }))
+            .await["result"];
+
+        let abort_blocks = devnet
+            .post_json(
+                "/abort_blocks".into(),
+                Body::from(json!({ "startingBlockHash": first_block["block_hash"] }).to_string()),
+            )
+            .await
+            .unwrap();
+
+        let aborted_blocks = get_json_body(abort_blocks).await;
+        assert_eq!(aborted_blocks["aborted"][0], second_block["block_hash"]);
+        assert_eq!(aborted_blocks["aborted"][1], first_block["block_hash"]);
+
+        let first_block_after_abort = &devnet
+            .send_custom_rpc(
+                "starknet_getBlockWithTxHashes",
+                json!({
+                    "block_id": {"block_hash": first_block["block_hash"]},
+                }),
+            )
+            .await["result"];
+        assert_eq!(first_block_after_abort["status"], "REJECTED".to_string());
+
+        let second_block_after_abort = &devnet
+            .send_custom_rpc(
+                "starknet_getBlockWithTxHashes",
+                json!({
+                    "block_id": {"block_hash": second_block["block_hash"]},
+                }),
+            )
+            .await["result"];
+        assert_eq!(second_block_after_abort["status"], "REJECTED".to_string());
+    }
+
+    #[tokio::test]
+    async fn abort_block_with_transaction() {
+        let devnet: BackgroundDevnet =
+            BackgroundDevnet::spawn_with_additional_args(&["--state-archive-capacity", "full"])
+                .await
+                .expect("Could not start Devnet");
+
+        let mint_hash = devnet.mint(FieldElement::ONE, 100).await;
+
+        let latest_block = &devnet
+            .send_custom_rpc("starknet_getBlockWithTxHashes", json!({ "block_id": "latest" }))
+            .await["result"];
+
+        let abort_blocks = devnet
+            .post_json(
+                "/abort_blocks".into(),
+                Body::from(json!({ "startingBlockHash": latest_block["block_hash"] }).to_string()),
+            )
+            .await
+            .unwrap();
+
+        let aborted_blocks = get_json_body(abort_blocks).await;
+        assert_eq!(aborted_blocks["aborted"][0], latest_block["block_hash"]);
+
+        let tx_status_after_abort =
+            devnet.json_rpc_client.get_transaction_status(mint_hash).await.unwrap();
+
+        assert_eq!(
+            tx_status_after_abort,
+            TransactionStatus::AcceptedOnL2(
+                starknet_rs_core::types::TransactionExecutionStatus::Reverted
+            )
+        );
```

Comment:
The existing `assert_tx_reverted` utility function from tests/common/utils.rs can be used

Reply:
Fixed in 94be08bbedfdbf021132c8b4d5574fdd327dc172

Reply:
Updated in 1c0ad1ead0f0e149ad681dc50028b6d0e756ead1